#784: Fix first entry not marked as read after sync

MrCustomizer opened the pull request:
This PR fixes the bug described in issue #784. Can it be merged or should this be handled differently?

David-Development commented:
Awesome, thank you so much for your Pull-Request! Very much appreciated!

I love the simplicity of your solution. It is definitely the right way to go! I think it might be even better to reset the index inside the `loadRssItemsIntoView` method so that it works also when the search is used.


In case you want to refactor some more code:

I also noticed that three lines further down the method `addOnItemTouchListener` is potentially called multiple times. Might make sense to always remove the (existing) touch listener before re-adding it. Not sure if you can actually add the same listener multiple times..? 

```java
recyclerView.removeOnItemTouchListener(itemTouchListener); // TODO does this crash?

if (rssItem.size() < 10) { // Less than 10 items in the list (usually 3-5 items fit on one screen)
   recyclerView.addOnItemTouchListener(itemTouchListener);
} 
```


Thank you! 😊

MrCustomizer commented:
Thanks for your fast response, David!
I've played a little bit around with the search, and you are exactly right. The bug appears there, too if you scroll down and up again, search for something and then scroll down again. The first item of the search result does not get marked as read in that case.
So I followed your advice and moved the new line to the method `loadRssItemsIntoView`, which solves this problem, too.

Regarding the `OnItemTouchListener`: The `RecyclerView` does not have an API to check for existing listeners. The listeners are internally saved in an ArrayList, where new listeners are just added to. And there is another member variable, where one of the listeners of the list is referenced All in all, the code in `RecyclerView` is a bit complicated, which is why I think it's better to be safe than sorry, here. I added some code to ensure it only gets added once.

MrCustomizer commented:
I just saw that your proposal for the refactoring was a little bit different than my approach. I've thought about that variant, too. But ultimately opted for the other one with the comment, because I thought it's more easily comprehensible. I'm not sure, though. Both ways seem to be a bit confusing to me. What do you think?

David-Development commented:
Awesome, thank you for the changes! Looks great to me! And thank you for the detailed explanation about the API. The API is not perfectly documented regarding this detail but I think by removing the listener manually and being "safe" is a good solution :) 

Thanks again for contributing! 👍😊

MrCustomizer commented:
My pleasure. Thanks for merging. :smiley:
